xarray/coding/variables.py

```diff
@@ -738,3 +779,37 @@ def encode(self, variable: Variable, name: T_Name = None) -> Variable:
 
     def decode(self, variable: Variable, name: T_Name = None) -> Variable:
         raise NotImplementedError()
+
+
+class LiteralTimedelta64Coder(VariableCoder):
+    """Code np.timedelta64 values."""
+
+    def encode(self, variable: Variable, name: T_Name = None) -> Variable:
+        if np.issubdtype(variable.data.dtype, np.timedelta64):
+            dims, data, attrs, encoding = unpack_for_encoding(variable)
+            resolution, _ = np.datetime_data(variable.dtype)
+            attrs["dtype"] = f"timedelta64[{resolution}]"
```

Comment:
What about _also_ including `units` in attrs?

That would make timedelta64 encoding still specify units in the style of CF conventions, which could make us a little more compatible with non-Xarray tools.

Reply:
Should be possible, but this would need an additional check inside `CFTimedeltaCoder` to prevent premature encoding and decoding if both attributes are attached.

Reply:
I agree this is a good idea, though does increase the complexity a little.  I gave it a try in my latest push.